src/bootstrap/check.rs

```diff
@@ -301,6 +301,61 @@ impl Step for CodegenBackend {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct RustAnalyzer {
+    pub target: TargetSelection,
+}
+
+impl Step for RustAnalyzer {
+    type Output = ();
+    const ONLY_HOSTS: bool = true;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.paths(&["src/tools/rust-analyzer", "rust-analyzer"])
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(RustAnalyzer { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) {
+        let compiler = builder.compiler(builder.top_stage, builder.config.build);
+        let target = self.target;
+
+        builder.ensure(Rustc { target });
```

Comment:
This doesn't look right, rust-analyzer doesn't use rustc_private.

Reply:
I copied that from the `tool_check_step!` macro (used for clippy, miri, rls, rustfmt, etc.)

Reply:
Right, those use rustc_private. Rust-analyzer doesn't.

Reply:
I'm looking at this again and, without it, I get some failures like:

```
error[E0463]: can't find crate for `core`

error[E0463]: can't find crate for `std`

error[E0463]: can't find crate for `compiler_builtins`

error[E0463]: can't find crate for `std`
 --> /home/amos/.cargo/registry/src/github.com-1ecc6299db9ec823/text-size-1.1.0/src/traits.rs:1:23
  |
1 | use {crate::TextSize, std::convert::TryInto};
  |                       ^^^ can't find crate

error: cannot find attribute `derive` in this scope
   --> /home/amos/.cargo/registry/src/github.com-1ecc6299db9ec823/drop_bomb-0.1.5/src/lib.rs:118:3
    |
118 | #[derive(Debug)]
    |   ^^^^^^
```

I think we need at least `ensure(Std {})`, maybe even `ensure(Rustc {})`

Reply:
Continued here: https://github.com/rust-lang/rust/pull/99603#issuecomment-1192629761